### PR TITLE
Fix refspec with prow pipeline when running on project-infra PRs

### PR DESCRIPTION
**Pull request by mboukhalfa**
It seems when we trigger a test from prow on project infra branch it fails because it cannot fetch the PR branch ! e.g https://jenkins.nordix.org/view/Metal3/job/metal3-e2e-clusterctl-upgrade-test-main/5/

This PR might help with this issue

### CODE REVIEW

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Review by kashifest — COMMENTED** [review]
Is this consistent with https://github.com/metal3-io/project-infra/blob/main/jenkins/jobs/bmo_e2e_tests.pipeline ?

File: jenkins/jobs/capm3-e2e-tests.pipeline [hunk 1]
```diff
@@ -12,6 +12,9 @@ script {
 
     if ("${PROJECT_REPO_ORG}" == 'metal3-io' && "${PROJECT_REPO_NAME}" == 'project-infra') {
         ci_git_branch = (env.PULL_PULL_SHA) ?: (env.ghprbActualCommit) ?: "main"
+        ci_git_base = (env.PULL_BASE_REF) ?: "main"
+        // Fetch the base branch and the pullSha when running on project-infra PR
```
Comment (kashifest) [inline jenkins/jobs/capm3-e2e-tests.pipeline:16]: you didnt use `pullSha`  here, but may be you should to make it consistent with other pipeline files ?

Comment (mboukhalfa) [reply jenkins/jobs/capm3-e2e-tests.pipeline]: thanks for noticing updated

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**